CriteoPublisherSdk/Sources/DataConsent/CR_GdprVersion.m

```diff
@@ -103,6 +122,54 @@ - (BOOL)isConsentGivenForPurpose:(NSUInteger)id {
   return purposeConsent != nil ? purposeConsent.boolValue : YES;
 }
 
+- (CR_GdprTcfPublisherRestrictionType)publisherRestrictionsForPurpose:(NSUInteger)id {
+  if (self.publisherRestrictionsKeyFormat == nil) {
+    return CR_GdprTcfPublisherRestrictionTypeNone;
+  }
+  NSString *purposeRestrictionKey =
+      [NSString stringWithFormat:self.publisherRestrictionsKeyFormat, id];
+  NSString *restrictionsForPurpose = [self.userDefaults stringForKey:purposeRestrictionKey];
+  NSUInteger criteoIndex = CR_CriteoTcfVendorID - 1;
```

Comment:
By including the offset in the constant initialization, you could save this line (unless the non-offset value is needed somewhere else?)

Reply:
Thanks! That comment made me realise that there was an off-by-one issue on the particular edge case checked there.

Fixed, which makes that variable to be removed.